web3deployment/deployerv3.js

```diff
@@ -0,0 +1,473 @@
+
+function parseInput( jsonInput ) {
+    const reserveData = jsonInput["reserves"];
+    const walletData = jsonInput["wallets"];
+
+    // reserve array
+    Object.values(reserveData).forEach(function(reserve) {
+      reserveDataArray.push(new Reserve(reserve));
+    });
+
+    // wallet array
+    Object.values(walletData).forEach(function(wallet) {
+      walletDataArray.push(new Wallet(wallet));
+    });
+
+    networkPermissions = jsonInput.permission["KyberNetwork"];
+    feeBurnerPermissions = jsonInput.permission["FeeBurner"];
+    wrapFeeBurnerPermissions = jsonInput.permission["WrapFeeBurner"];
+    expectedRatePermissions = jsonInput.permission["ExpectedRate"];
+
+    maxGasPrice = jsonInput["max gas price"].toString();
+    negDiffInBps = jsonInput["neg diff in bps"].toString();
+    minExpectedRateSlippage = jsonInput["min expected rate slippage"].toString();
+    defaultWalletFeesBps = jsonInput["default wallet fees bps"].toString();
+    taxFeesBps = jsonInput["tax fees bps"].toString();
+    taxWalletAddress = jsonInput["tax wallet address"];
+    initialKncToEthRatePrecision = jsonInput["KNC to ETH rate"].toString();
+
+    dgxTokenAddress = jsonInput["addresses"].dgx;
+    kncTokenAddress = jsonInput["addresses"].knc;
+    wethTokenAddress = jsonInput["addresses"].weth;
+    medianizerAddress = jsonInput["addresses"].medianizer;
+    factoryAddress = jsonInput["addresses"].factory;
+    proxyAddress = jsonInput["addresses"].proxy;
+    whitelistAddress = jsonInput["addresses"].whitelist;
+    feeBurnerWrapperProxyAddress = jsonInput["addresses"].feeBurnerWrapperProxy;
+
+    // output file name
+    outputFileName = jsonInput["output filename"];
+};
+
+async function setPermissions(contract, permJson) {
+  console.log("set operator(s)");
+  for(let i = 0 ; i < permJson.operator.length ; i++ ) {
+    const operator = permJson.operator[i];
+    console.log(operator);
+    await sendTx(contract.methods.addOperator(operator));
+  }
+  console.log("set alerter(s)");
+  for(let i = 0 ; i < permJson.alerter.length ; i++ ) {
+    const alerter = permJson.alerter[i];
+    console.log(alerter);
+    await sendTx(contract.methods.addAlerter(alerter));
+  }
+  console.log("transferAdminQuickly");
+  const admin = permJson.admin;
+  console.log(admin);
+  await sendTx(contract.methods.transferAdminQuickly(admin));
+}
+
+
+async function main() {
+  nonce = await web3.eth.getTransactionCount(sender);
+  console.log("nonce",nonce);
+
+  chainId = chainId || await web3.eth.net.getId()
+  console.log('chainId', chainId);
+
+  console.log("starting compilation");
+  const output = await solc.compile({ sources: input }, 1);
+  // console.log(output);
+  console.log("finished compilation");
+
+  if (!dontSendTx) {
+    await waitForEth();
+  }
+
+  console.log("deploying kyber network");
+  [networkAddress,networkContract] = await deployContract(output, "KyberNetwork.sol:KyberNetwork", [sender]);
```

Comment:
after each deploy please print address as output.
since if deploy script fails in middle of deploy, its easier to restart a deploy and reuse some of the deployed contracts.

Reply:
Ah, so u want it after every deploy, ok.